src/components/forms/SelectField/SelectField.stories.mdx

```diff
@@ -0,0 +1,96 @@
+import {Canvas, Meta, Story} from '@storybook/addon-docs';
+import {expect} from '@storybook/jest';
+import {userEvent, within} from '@storybook/testing-library';
+
+import {FormikDecorator} from 'story-utils/decorators';
+
+import AsyncSelectField from './AsyncSelectField';
+import SelectField from './SelectField';
+
+export const Template = args => <SelectField {...args} />;
+export const DynamicTemplate = args => <AsyncSelectField {...args} />;
+const retrieveOptions = async () => {
+  await new Promise(resolve => setTimeout(resolve, 1000));
+  return [
+    {value: 'option1', label: 'Option 1'},
+    {value: 'option2', label: 'Option 2'},
+    {value: 'option3', label: 'Option 3'},
+  ];
+};
+const test = async canvasElement => {
+  const canvas = within(canvasElement);
+  await expect(canvas.getByRole('combobox')).toBeVisible();
+  await expect(canvas.getByText('Option')).toBeVisible();
+  await expect(canvas.getByText('This is a custom description for the option field')).toBeVisible();
+  const label = canvas.getByText('Option');
+  userEvent.click(label);
+  await expect(canvas.getByRole('combobox')).toHaveFocus();
+};
+
+<Meta
+  title="Pure React Components / Forms / SelectField"
+  component={SelectField}
+  parameters={{
+    initialValues: {
+      test: 'test',
+    },
+  }}
+/>
+
+# Select Field
+
+We should make use of the SelectField component whenever we have `options` that are statically
+available, and we don't need to retrieve them from an API endpoint. Whenever we **do** need to
+retrieve the options from an API endpoint, we should instead use the AsyncSelectField component.
```

Comment:
```suggestion
Use the `SelectField` component whenever you have `options` that are known upfront. If your
options are retrieved from some external data source (like an API endpoint) instead, use the 
`AsyncSelectField` variant instead.
```

- address the reader
- imperative form makes it easier to extract what to do when, as a reader